java/src/jmri/jmrix/loconet/LocoNetSlot.java

```diff
@@ -40,24 +43,69 @@ public class LocoNetSlot {
      * @param slotNum  slot number to be assigned to the new LocoNetSlot object
      */
     public LocoNetSlot(int slotNum) {
+        this(slotNum, LnConstants.LOCONETPROTOCOL_UNKNOWN);
+    }
+
+    /**
+     * Create a slot based solely on a slot number.  The remainder of the slot is
+     * left un-initialized.
+     * <p>
+     * @param slotNum - slot number to be assigned to the new LocoNetSlot object
+     * @param inLoconetProtocol - can be 0 = unknown, 1 = version 1.1, 2 = Expandedslot
+     */
+    public LocoNetSlot(int slotNum, int inLoconetProtocol) {
+        this(slotNum, inLoconetProtocol, SlotType.LOCO);
+        if ((slotNum == 0) || (slotNum > 120 && slot < 128)
+                || (slotNum > 247 && slotNum < 257)
+                || (slotNum > 375 && slotNum < 385)) {
+            slotType = SlotType.SYSTEM;
+        } else {
+            slotType = SlotType.LOCO;
+        }
+    }
+
+    /**
+     * Create a slot , initialize slotnum, protocol and slot type
+     * <p>
+     * @param slotNum - slot number to be assigned to the new LocoNetSlot object
+     * @param inLoconetProtocol - can be 0 = unknown, 1 = version 1.1, 2 = Expandedslot
+     * @param inSlotType - SLotType enum
+     */
+    public LocoNetSlot(int slotNum, int inLoconetProtocol, SlotType inSlotType) {
         slot = slotNum;
+        loconetProtocol = inLoconetProtocol;
+        if (slotNum > 127 ) {
+            // has to be 2
+            loconetProtocol = LnConstants.LOCONETPROTOCOL_TWO;
```

Comment:
If loconetProtocol is not 2, it might be better to throw an exception here, or at least log an error message to the log. It makes it easier to find potential bugs.

Reply:
FYI - I believe that throwing an exception could be bad in some cases.  For example, for some command stations, it can be useful to look at "aliased" slots even though the command station only supports 100ish slots.  the "aliased" slots.

Reply:
I see.

@Pugwash1 
You can ignore my request.